flask_security/forms.py

```diff
@@ -125,7 +125,8 @@ class NewPasswordFormMixin():
 class PasswordConfirmFormMixin():
     password_confirm = PasswordField(
         get_form_field_label('retype_password'),
-        validators=[EqualTo('password', message='RETYPE_PASSWORD_MISMATCH')])
+        validators=[password_required,
```

Comment:
But one may give the password yet forget to reconfirm it. In this case, a request can still be posted to the server. By providing a data required attributes, the browser should remind the users that the reconfirm password field is empty. This provides better user experience and save server resources by filtering invalid posts before sending.